lua/neo-tree/sources/common/file-nesting.lua

```diff
@@ -25,30 +57,224 @@ function M.get_parent(item)
+  local test = str:gsub("\\[%*%?%[%]]", "")
+  local pos, _ = test:find("*")
+  if pos ~= nil then
+    return true
+  end
+  return false
+end
+
+local function case_insensitive_pattern(pattern)
+  -- find an optional '%' (group 1) followed by any character (group 2)
+  local p = pattern:gsub("(%%?)(.)", function(percent, letter)
+    if percent ~= "" or not letter:match("%a") then
+      -- if the '%' matched, or `letter` is not a letter, return "as is"
+      return percent .. letter
+    else
+      -- else, return a case-insensitive character class of the matched letter
+      return string.format("[%s%s]", letter:lower(), letter:upper())
+    end
+  end)
+
+  return p
 end
 
---- Checks if `target` should be nested into `base`
+function table_is_empty(table_to_check)
+  return table_to_check == nil or next(table_to_check) == nil
+end
+
+function flatten_nesting(nesting_parents)
+  for key, config in pairs(nesting_parents) do
+    if config.is_nested ~= nil then
+      local parent = config.nesting_parent
+      -- count for emergency escape
+      local count = 0
+      while parent.nesting_parent ~= nil and count < 100 do
+        parent = parent.nesting_parent
+        count = count + 1
+      end
+      if parent ~= nil then
+        for _, child in pairs(config.children) do
+          child.nesting_parent = parent
+          table.insert(parent.children, child)
+        end
+        config.children = nil
+      end
+    end
+    nesting_parents[key] = nil
+  end
+end
+
+function M.nest_items(context)
+  if M.is_enabled() == false or table_is_empty(context.nesting) then
+    return
+  end
+  for _, config in pairs(context.nesting) do
+    local files = config.nesting_callback(config, context.all_items)
+    local folder = context.folders[config.parent_path]
+    for _, to_be_nested in ipairs(files) do
+      table.insert(config.children, to_be_nested)
+      to_be_nested.is_nested = true
+      to_be_nested.nesting_parent = config
+      if folder ~= nil then
+        for index, file_to_check in ipairs(folder.children) do
+          if file_to_check.id == to_be_nested.id then
+            table.remove(folder.children, index)
+          end
+        end
+      end
+    end
+  end
+
+  flatten_nesting(context.nesting)
+end
+
+--- Returns `item` nesting parent path if exists
+---@return string?
+function get_parent(item, siblings)
+  if item.type ~= "file" then
+    return nil
+  end
+  for _, matcher in pairs(matchers) do
+    if matcher.enabled then
+      local parent = matcher.get_parent(item, siblings)
+      if parent ~= nil then
+        return parent
+      end
+    end
+  end
+
+  return nil
+end
+
+--- Checks if `item` have a valid nesting lookup
 ---@return boolean
-function M.should_nest_file(base, target)
-  local ext_lookup = M.config[base.exts]
+function M.can_have_nesting(item)
+  for _, matcher in pairs(matchers) do
+    if matcher.enabled then
+      if matcher.can_have_nesting(item) then
+        return
```

Comment:
Should this be `return true`?

Reply:
You are right, this is legacy code which was replaced by get_nesting_callback. I will remove it.